rust-install/src/component/components.rs

```diff
@@ -36,22 +40,23 @@ impl<'a> ChangeSet<'a> {
 	}
 }
 
-#[derive(Clone)]
+#[derive(Clone, Debug)]
 pub struct Components {
 	prefix: InstallPrefix,
 }
 
 impl Components {
-	pub fn new(prefix: InstallPrefix) -> Option<Self> {
-		if utils::is_file(prefix.manifest_file(COMPONENTS_FILE)) {
-			Some(Components { prefix: prefix })
-		} else {
-			None
-		}
-	}
-	pub fn init(prefix: InstallPrefix) -> Result<Self> {
-		try!(utils::write_file("components", &prefix.manifest_file(COMPONENTS_FILE), ""));
-		Ok(Components { prefix: prefix })
```

Comment:
Huh, I thought I left a comment about this. It's because initializing the `components` file before any components are written, and outside of a transaction, could leave the installation in a partial state where nothing is installed but there is a rustlib/components file. Both 'components' and 'rust-installer-version' are initialized inside the transaction that installs the first component.